addons/account/models/account_bank_statement.py

```diff
@@ -564,12 +352,35 @@ class AccountBankStatementLine(models.Model):
         string='Auto-generated Payments',
         help="Payments generated during the reconciliation of this bank statement lines.")
 
+    # == Technical fields ==
+    internal_index = fields.Char(
+        string='Internal Reference',
+        compute='_compute_internal_index',
+        store=True,
+        index=True,
+        help="Technical field used to store the internal reference of the statement line for fast indexing."
+    )
     # == Display purpose fields ==
-    is_reconciled = fields.Boolean(string='Is Reconciled', store=True,
-        compute='_compute_is_reconciled',
+    is_reconciled = fields.Boolean(string='Is Reconciled',
+        compute='_compute_is_reconciled', store=True,
         help="Technical field indicating if the statement line is already reconciled.")
-    state = fields.Selection(related='statement_id.state', string='Status', readonly=True)
+    statement_state = fields.Selection(
+        string='Statement Status',
+        related='statement_id.state', readonly=True, store=True
+    )
+    statement_is_valid = fields.Boolean(
+        related='statement_id.is_valid', readonly=True, store=False
+    )
     country_code = fields.Char(related='company_id.account_fiscal_country_id.code')
+    # Not the values of this field does not necessarily correspond to the cumulated balance in the account move line.
+    # here these values correspond to occurrence order (the reality) and the should match the bank report
+    # but in the move lines, it corresponds to the recognition order. But these two values should probably be the same
+    # at the end of each day.
```

Comment:
Please check the formulation

Reply:
It was about the date, currently, the lock date algorithm can lead to have a different date, on the moves and in here, we used to have another date in here, but now they are the same. (to many spec changes!)
But anyway, this running balance is using checkpoints, and tries to stick into the bank data not the company data.